### PR TITLE
Fix data-store DB job load

### LINKED ISSUE #4784: data store: issues when reloading workflows
Some odd behaviour spotted when reloading workflows.

Start off with a regular workflow and a few job submissions:

Screenshot from 2022-03-29 16-50-43

Then reload the workflow and old jobs disappear from the data store:

Screenshot from 2022-03-29 16-51-02

Now trigger the task again, the oldest job still in the store goes from `failed` to `submit-failed`, stranger still all subsequent jobs start as `submit-failed`, then go through `running` to `failed`:

Screenshot from 2022-03-29 16-51-24

See also #4036

```ini
[scheduler]
    initial cycle point = 1
    cycling mode = integer
    [[graph]]
        R1 = foo => bar
[runtime]
    [[foo,bar]]
        script = false
```

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

### PR DESCRIPTION
**Pull request by dwsutherland**
These changes close #4784

For example, with triggering an historical task:
```
$ cylc trigger linear //20210801T00/qux
```

**Before**
(only one job loaded from DB)
image

**After**
image

Also with the window about a historical task:
```
$ cylc trigger linear //20210801T00/qaz
```
**After**
image

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Test added to cover DB load of jobs on restart.
- [x] Appropriate change log entry included.
- [x] No documentation update required.

### CODE REVIEW

**Comment by dwsutherland** [comment]
have to figure out another couple of issues:
- when restarting under certain circumstances the job history is not there
- when restarted with historical job failures the state is incorrect..

Looking into it now..

**Comment by dwsutherland** [comment]
> have to figure out another couple of issues:
> 
> * when restarting under certain circumstances the job history is not there
> * when restarted with historical job failures the state is incorrect..
> 
> Looking into it now..

Ok fixed.

File: CHANGES.md [hunk 1]
```diff
@@ -85,6 +85,9 @@ workflow source argument), and rename the `--flow-name` option to
 
 ### Fixes
 
+[#4891](https://github.com/cylc/cylc-flow/pull/4891) - Fix bug that could cause
+past jobs to be omitted in the UI.
+
```
Comment (MetRonnie) [inline CHANGES.md:90]: Entry should be under rc4 not rc3

File: cylc/flow/rundb.py [hunk 2]
```diff
@@ -647,11 +647,12 @@ def select_jobs_for_restart(self, callback):
...
                 %(task_pool)s.name,
-                %(task_pool)s.status,
-                %(task_states)s.submit_num,
+                %(task_jobs)s.submit_num,
                 %(task_jobs)s.time_submit,
+                %(task_jobs)s.submit_status,
                 %(task_jobs)s.time_run,
                 %(task_jobs)s.time_run_exit,
+                %(task_jobs)s.run_status,
                 %(task_jobs)s.job_runner_name,
```
Comment (MetRonnie) [inline cylc/flow/rundb.py:656]: The docstring is now out of date: https://github.com/cylc/cylc-flow/blob/6613349ed1069ef08b7512c81e29b23b4d4e9731/cylc/flow/rundb.py#L642-L645

Possibly just remove that list in the docstring as it's duplication of what can be read from the SQL query anyway

**Review by MetRonnie — COMMENTED** [review]
Tested out and seems to do the trick. Just a couple of niggles

**Review by hjoliver — APPROVED** [review]
LGTM @dwsutherland :+1:   Can be merged once you've addressed @MetRonnie 's comments.

**Comment by dwsutherland** [comment]
> LGTM @dwsutherland 👍 Can be merged once you've addressed @MetRonnie 's comments.

Addressed.

**Pull request merged**